packages/key-management/src/KeyAgentBase.ts

```diff
@@ -57,7 +60,18 @@ export abstract class KeyAgentBase implements KeyAgent {
     return this.#account.deriveAddress(paymentKeyDerivationPath, stakeKeyDerivationIndex);
   }
 
+  async deriveSharedWalletAddress({
```

Comment:
```suggestion
```
There's no need to duplicate the functionality for key/address derivation as it's exactly the same as for regular wallets. Instead, add a `purpose` parameter to the constructor of KeyAgent.

Reply:
Removed on fac71bd